crates/bevy_pbr/src/light_probe/environment_filter.wgsl

```diff
@@ -0,0 +1,427 @@
+#import bevy_render::maths::{PI, PI_2};
+#import bevy_pbr::lighting::perceptualRoughnessToRoughness;
+
+struct PrefilterConstants {
+    mip_level: f32,
+    sample_count: u32,
+    roughness: f32,
+    blue_noise_size: vec2f,
+}
+
+@group(0) @binding(0) var input_texture: texture_2d_array<f32>;
+@group(0) @binding(1) var input_sampler: sampler;
+@group(0) @binding(2) var output_texture: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(3) var<uniform> constants: PrefilterConstants;
+@group(0) @binding(4) var blue_noise_texture: texture_2d<f32>;
+
+// Tonemapping functions to reduce fireflies
+
+const white_point: f32 = 1.0;
+fn rcp(x: f32) -> f32 { return 1.0 / x; }
+fn max3(x: vec3f) -> f32 { return max(x.r, max(x.g, x.b)); }
+fn tonemap(color: vec3f) -> vec3f {
+    return color / (color + vec3(white_point));
+}
+fn reverse_tonemap(color: vec3f) -> vec3f {
+    return white_point * color / (vec3(1.0) - color);
+}
+
+// Predefined set of uniform directions
+fn get_uniform_direction(index: u32) -> vec3f {
+    var dir = vec3f(0.0);
+    
+    switch(index % 64u) {
```

Comment:
Array indexing or even indexing into a buffer might make more sense here? Idk.

Reply:
Removed the switch statement in favor of golden angle based dynamic calculation and a new function called uniform_sample_sphere